coderd/promoauth/oauth2.go

```diff
@@ -70,13 +130,53 @@ func (f *Factory) New(name string, under OAuth2Config) *Config {
 	}
 }
 
+// NewGithub returns a new instrumented oauth2 config for github. It tracks
+// rate limits as well as just the external request counts.
+//
+//nolint:bodyclose
+func (f *Factory) NewGithub(name string, under OAuth2Config) *Config {
+	cfg := f.New(name, under)
+	cfg.interceptors = append(cfg.interceptors, func(resp *http.Response, err error) {
+		limits, ok := githubRateLimits(resp, err)
+		if !ok {
+			return
+		}
```

Comment:
Yup exactly. In prod, I think this would be a niche edge case like some http gateway error or something.

The idea though is to not return incorrect information. So if the headers are not present, I'd rather it be a noop than set the limits to `0`, only for the next request to bump them back to actual values.